packages/gatsby-remark-copy-linked-files/src/index.js

```diff
@@ -149,6 +149,11 @@ module.exports = (
       return
     }
 
+    // since dir will be undefined on non-files
+    if (getNode(markdownNode.parent).internal.type !== `File`) {
```

Comment:
You need to check if there is a parent as well since a markdown node doesn't necessarily need a parent.